lms/views/admin/__init__.py

```diff
@@ -6,7 +6,7 @@
 
 @forbidden_view_config(path_info="/admin/*")
 def forbidden(request):
-    if request.identity.userid:
+    if request.identity and request.identity.userid:
```

Comment:
This is the only place I could find that needed tweaking. Not suprisingly is in a forbidden view.

General views shouldn't haven have to worry about having or not an identity, we use the identity permission for that.